### PR TITLE
Fix unnecessary GlobalSearchField rerender

### PR DESCRIPTION
**Pull request by sri2k1us**
- Misc attempts to fix an error in `terrain.js`.

### CODE REVIEW

**Review by johnworth — APPROVED** [review]
LGTM. :shipit:

File: src/components/utils/useWhyDidYouUpdate.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            // If changesObj not empty then output to console
+            if (Object.keys(changesObj).length) {
+                console.log("[why-did-you-update]", name, changesObj);
+            }
+        }
+
+        // Finally update previousProps with current props for next hook call
+        previousProps.current = props;
+    });
+}
```
Comment (slr71) [inline src/components/utils/useWhyDidYouUpdate.js:35]: This seems incredibly useful. 👍

**Review by slr71 — APPROVED** [review]
LGTM :shipit:

**Comment by sri2k1us** [comment]
Thanks for the reviews.

File: src/components/utils/useWhyDidYouUpdate.js (hunk 1 above)
Comment (psarando) [reply src/components/utils/useWhyDidYouUpdate.js:35]: Yes, this will come in handy 👍

**Pull request merged**